### ISSUE TITLE
Can the state machine be compiled directly into target language rather than to structures interpreted by interpreter?

### ISSUE THREAD

**Issue by KOLANICH** [waxeye-org/waxeye#100]

**Comment by orlandohill** [comment]
Yes. If I understand correctly, any language that can be interpreted can also be compiled, and vice versa. The simplest way would be to have the compiler traverse the state machine and output the same code that the interpreter uses.

It's something that I had in mind to do later, but only as an optimization once more important features were done.
